File: .yalc/@entur/atb-mobile-client-sdk/token/token-core-android-lib/src/main/java/no/entur/abt/android/token/keystore/KeystoreAliasFactory.java
package no.entur.abt.android.token.keystore;

/**
 * 
 * Key alias factory. Format: <br>
 * <br>
 * 1. global prefix<br>
 * 2. token context<br>
 * 3. token id<br>
 * 4. key<br>
 */

public interface KeystoreAliasFactory {

	String getAlias(String tokenContextId, String tokenId, String key);

	String getTokenAliasPrefix(String tokenContextId, String tokenId);

	String getContextAliasPrefix(String tokenContextId);

	String getGlobalAliasPrefix();

	String getEncryptionKeyAlias(String tokenContextId, String tokenId);

	String getSignatureKeyAlias(String tokenContextId, String tokenId);

}

File: .yalc/@entur/atb-mobile-client-sdk/token/token-core-ios-lib/AbtTokenCoreSDK/keystore/KeyStoreAliasFactory.swift
import Foundation

public protocol KeyStoreAliasFactory {
    func getAlias(tokenContext: TokenContext, tokenId: String, alias: String) -> String

    func getTokenAliasPrefix(tokenContext: TokenContext, tokenId: String) -> String

    func getContextAliasPrefix(tokenContext: TokenContext) -> String

    func getGlobalAliasPrefix() -> String

    func getEncryptionKeyAlias(tokenContext: TokenContext, tokenId: String) -> String

    func getSignatureKeyAlias(tokenContext: TokenContext, tokenId: String) -> String

    func getSignatureCertificateKeyAlias(tokenContext: TokenContext, tokenId: String) -> String
}
